src/backends/plonky2/circuits/mainpod.rs

```diff
@@ -360,6 +473,41 @@ impl OperationVerifyGadget {
         builder.all([op_code_ok, st_code_ok, arg_prefix_ok, no_dupes_ok])
     }
 
+    fn eval_lt_to_neq(
+        &self,
+        builder: &mut CircuitBuilder<F, D>,
+        st: &StatementTarget,
+        op: &OperationTarget,
+        resolved_op_args: &[StatementTarget],
+    ) -> BoolTarget {
+        let op_code_ok = op.has_native_type(builder, NativeOperation::LtToNotEqual);
+
+        let st_code_ok = st.has_native_type(builder, &self.params, NativePredicate::NotEqual);
+
+        let arg_type_ok =
+            resolved_op_args[0].has_native_type(builder, &self.params, NativePredicate::Lt);
+
+        let arg1_key = resolved_op_args[0].args[0].clone();
+        let arg2_key = resolved_op_args[0].args[1].clone();
+
+        let expected_st_args: Vec<_> = [arg1_key, arg2_key]
+            .into_iter()
+            .chain(std::iter::repeat_with(|| StatementArgTarget::none(builder)))
+            .take(self.params.max_statement_args)
+            .flat_map(|arg| arg.elements)
+            .collect();
+
+        let st_args_ok = builder.is_equal_slice(
```

Comment:
This is a nitpick but since this check only considers 1 possible statement code, why not check the expected statement like this?
```rust
        let expected_statement = StatementTarget::new_native(
            builder,
            &self.params,
            NativePredicate::NotEqual,
            &[arg1_key1.clone(), arg2_key2.clone()],
        );
        let st_ok = builder.is_equal_flattenable(st, &expected_statement);
```
And get rid of the check 
```rust
let st_code_ok = st.has_native_type(builder, &self.params, NativePredicate::NotEqual);
```